Jropp/jason-ropp-js-iobp
# Onboard Project: Utils.js Refactor

## What It Does

Do you wish that your utils.js had properly separated and simplified code? Well so did I!

Through the course of the project my skills for keeping code clean have improved. This code had some particularly complicated spots that are still a little hairy but new skills has made them somewhat more navigable for future maintainers/reviewers.

Here's the lowdown. The biggest change here is that`insertPhoneSymbols` function was getting a little rambunctious, so I went ahead and hacked it into three smaller parts:

1. A parent `formatPhoneNumber()`
2. A child `insertDomesticSymbols()`
3. An `insertInternationalSymbols()` 

Also, some of the naming was just confusing, (like why would you name your child Apple confusing) so I made some little tweaks as well.

## How To Test

**If you can find a way new to break it, I'll give you a dollar.** 

- Create, edit, delete users to ensure functionality was left intact.

- Try to find ways to break the phone input. More than 13 numbers and less than 10 should give you a warning message. Any other version of the numbers should display properly once the user is saved.

```
Format: +xxx (xxx) xxx-xxxx
```

- Look through the code and check for instances where:

      - Naming is unclear
      - Code should be separated into another function
      - Variables or steps are unnecessary (no functional value, do not aid readability)

## Notes/Caveats

This does not address the smell coming from my kitchen at the moment.

## Checklist

Under penalty of public shaming, I avow that I:


Browsers tested:
- [ ] ~Internet Explorer~<--Aint nobody got time for that

## Dependencies

- [ ] nu nu nu nada
